internal/inspectimage/writer/structured_bom_format_test.go

```diff
@@ -0,0 +1,100 @@
+package writer_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/buildpacks/pack/internal/inspectimage"
+
+	"github.com/buildpacks/pack/internal/inspectimage/writer"
+
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/pack"
+	"github.com/buildpacks/pack/internal/config"
+	ilogging "github.com/buildpacks/pack/internal/logging"
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+func TestStructuredBOMFormat(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+	spec.Run(t, "StructuredBOMFormat Writer", testStructuredBOMFormat, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testStructuredBOMFormat(t *testing.T, when spec.G, it spec.S) {
+	var (
+		assert = h.NewAssertionManager(t)
+		outBuf bytes.Buffer
+
+		remoteInfo *pack.ImageInfo
+		localInfo  *pack.ImageInfo
+	)
+
+	when("Print", func() {
+		it.Before(func() {
+			remoteInfo = &pack.ImageInfo{}
+			localInfo = &pack.ImageInfo{}
+			outBuf = bytes.Buffer{}
+		})
+
+		// Just test error cases, all error-free cases will be tested in JSON, TOML, and YAML subclasses.
+		when("failure cases", func() {
+			when("both info objects are nil", func() {
+				it("displays a 'missing image' error message'", func() {
+					sharedImageInfo := inspectimage.GeneralInfo{
+						Name:            "missing-image",
+						RunImageMirrors: []config.RunImage{},
+					}
+
+					structuredBOMWriter := writer.StructuredBOMFormat{
+						MarshalFunc: testMarshalFunc,
+					}
+
+					logger := ilogging.NewLogWithWriters(&outBuf, &outBuf)
+					err := structuredBOMWriter.Print(logger, sharedImageInfo, nil, nil, nil, nil)
+					assert.ErrorWithMessage(err, fmt.Sprintf("unable to find image '%s' locally or remotely", "missing-image"))
```

Comment:
This seems like a slightly different assertion than we were doing. Is this output still being generated? 
https://github.com/buildpacks/pack/pull/957/files#diff-2dbbb729dc74480a1e93e56403bb3246e3e81ba9ba7ae8b0e4486ff321ec45eeL71-L79
```
h.AssertEq(t,
						outBuf.String(),
						`{"remote":null,"local":null}`+"\n")
```

Reply:
Ya this is different behavior. Could revert if we want to keep a `pack` consistent. But it does feel awkward having `pack inspect-builder <image-name> --bom` succeed and return an empty `json` array when we failed to find relevant images.

Reply:
Ya this is different behavior. Could revert if we want to keep a `pack` consistent. But it does feel awkward having `pack inspect-builder <image-name> --bom` succeed and return an empty `json` array when we failed to find relevant images.